lib/src/game/viewport.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:math' as math;
+import 'dart:ui';
+
+import 'package:flutter/painting.dart';
+
+import '../../game.dart';
+import '../../extensions.dart';
+import '../../palette.dart';
+
+abstract class Viewport {
+  void resize(Vector2 newRawSize);
+  void render(Canvas c, void Function(Canvas c) renderGame);
+  Vector2 getEffectiveSize();
+  Vector2 getRawSize();
+}
+
+class DefaultViewport extends Viewport {
+  Vector2 _size;
+
+  @override
+  void render(Canvas c, void Function(Canvas c) renderGame) {
+    renderGame(c);
+  }
+
+  @override
+  void resize(Vector2 newRawSize) {
+    _size = newRawSize;
+  }
+
+  @override
+  Vector2 getEffectiveSize() => _size;
+
+  @override
+  Vector2 getRawSize() => _size;
+}
+
+class FixedRatioViewport extends Viewport {
```

Comment:
Maybe some short explanations of each viewport

Reply:
totally forgot!